File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void inorder(TreeNode root, int k,int[] ans,int[] c){
        if(root.left!=null)  inorder(root.left,k,ans,c);

        c[0]=c[0]+1;
        if(k==c[0]) ans[0]=root.val;
        if(root.right!=null) inorder(root.right,k,ans,c);
    }
    public int kthSmallest(TreeNode root, int k) {
        int[] ans=new int[1];
        int[] c=new int[1];
        inorder(root,k,ans,c);
        return ans[0];
    }
}

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def kthSmallest(self, root, k):
        """
        :type root: TreeNode
        :type k: int
        :rtype: int
        """
        def check(node, ans, an):
            if node == None:
                return None
            
            check(node.left, ans, an)
            ans[0] += 1
            if ans[0] == k:
                an.append(node.val)
            check(node.right, ans, an)
            
            return node


        ans = [0]
        an = []
        check(root, ans, an)
        return an[0]
        